### PR TITLE
[fix] rrelu, rrelu_, & RReLU when lower bound > upper bound

### LINKED ISSUE #83160: torch.nn.functional.rrelu_ allows lower bound greater than upper bound
### 🐛 Describe the bug

torch.nn.functional.rrelu_ allows lower bound greater than upper bound. It doesn't throw exception, and the document has no 'add note' for case lower bound> upper bound.

```
import torch
results={}
arg_1 = torch.rand([2], dtype=torch.float64)
arg_2 = 2
arg_3 = 0
results['res'] = torch.nn.functional.rrelu_(arg_1,arg_2,arg_3,)
```

Above code works.

### Versions

pytorch: 1.8.1
python: 3.8.3
os: win11

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

### PR DESCRIPTION
**Pull request by khushi-411**
Fixes #83160

cc @kshitij12345 @albanD

### CODE REVIEW

File: aten/src/ATen/native/Activation.cpp [hunk 1]
```diff
@@ -599,10 +599,12 @@ Tensor rrelu_with_noise_backward(
 }
 
 Tensor rrelu(const Tensor & self, const Scalar& lower, const Scalar& upper, bool training, c10::optional<Generator> generator) {
+  TORCH_CHECK(lower.to<double>() < upper.to<double>(), "Lower bound must be less than the upper bound")
```
Comment (mruberry) [inline aten/src/ATen/native/Activation.cpp:602]: Why not weakly less? `uniform_()` allows the lower and upper bounds of the distribution to be the same

File: test/test_nn.py [hunk 2]
```diff
@@ -14500,6 +14500,24 @@ def test(nonlinearity, *args, **kwargs):
...
+                elif input is None:
+                    module(lower, upper, inplace)(torch.rand((2)))
+                module(input, lower, upper, inplace)
+
+        input = torch.rand((3, 3), device=device, dtype=torch.float32)
+        lower, upper = 0.4, 0.1
+
+        test(False, nn.RReLU, None, lower, upper)
+        test(True, nn.RReLU, None, lower, upper)
+        test(False, nn.functional.rrelu, input, lower, upper)
```
Comment (mruberry) [inline test/test_nn.py:14517]: Can this test just be an ErrorInput for `nn.functional.rrelu`? The module just calls the function:

https://github.com/pytorch/pytorch/blob/ebd4e90ff7f6d1e57bbf6f2e717a8addb6b28e28/torch/nn/modules/activation.py#L167

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -2320,6 +2320,11 @@ def error_inputs_gradient(op_info, device, **kwargs):
                          error_type=RuntimeError,
                          error_regex='torch.gradient expected each dimension size to be at least')
 
+def error_inputs_rrelu(op_info, device, **kwargs):
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:2323]: Nice addition

**Review by mruberry — COMMENTED** [review]
Hey @khushi-411!

This is a fine addition that is, in spirit, consistent with PyTorch's handling of the `uniform_` operation. However, I think we should try to be completely consistent (since in both cases we are defining a uniform distribution), and allow for "weakly less." 

Also see inline comment for whether the test update should just be an ErrorInput.

File: aten/src/ATen/native/Activation.cpp (hunk 1 above)
Comment (khushi-411) [reply aten/src/ATen/native/Activation.cpp]: Hi, @mruberry! Thanks for pointing it out. This makes sense to me! Have updated it.

File: torch/testing/_internal/common_methods_invocations.py (hunk 3 above)
Comment (khushi-411) [reply torch/testing/_internal/common_methods_invocations.py:2253]: Thanks!

File: test/test_nn.py (hunk 2 above)
Comment (khushi-411) [reply test/test_nn.py]: Sure, removed these tests in favor of ErrorInputs.

**Comment by khushi-411** [comment]
Hi, @mruberry! Gentle ping :)

**Review by mruberry — APPROVED** [review]
Cool! Thanks @khushi-411!

**Comment by khushi-411** [comment]
Thanks for reviewing, @mruberry!!

**Review by albanD — APPROVED** [review]

**Comment by khushi-411** [comment]
Thanks, @albanD!

**Pull request closed without merging**